### PR TITLE
Tmedia 724 header chain - subtask: tmedia 886

### PR DESCRIPTION
**Pull request by brentmiller1973**
## Description

This is second subtask listed in https://arcpublishing.atlassian.net/browse/TMEDIA-724 for the header nav chain work.  The goal is of this ticket is to make a pass and remove all dependence to engine, news-css, and shared styles and replace them with Arc components and theme styles where appropriate. 

**Note: I am creating this  pr that on approval will be merged into the base feature branch for this work.**

## Jira Ticket

- Sub Task: https://arcpublishing.atlassian.net/browse/TMEDIA-886

## Acceptance Criteria

- Remove all engine-sdk dependencies used and translate the css into advanced styling.
- Refactor code from engine-sdk components to use Arc components.
- Remove dependencies from shared styles and replace with arc components or custom.
- Determine if any engine-sdk utility code should be moved to the utils directory of the components repo.

## Test Steps

1. Checkout this branch `git checkout tmedia-724-header-chain-base-feature-branch`
2. Run storybook
3. Compare the local storybook stories with a recent published storybook (such as https://5f91e4721ccaba0022a10782-nyjbszewjr.chromatic.com/?path=/story/chains-header-nav-block--wide-logo )

### CODE REVIEW

File: blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/horizontal-links/default.jsx [hunk 1]
```diff
@@ -39,6 +38,23 @@ const HorizontalLinksBar = ({ hierarchy, navBarColor, showHorizontalSeperatorDot
...
+		const formattedURL = formatURL(linkUrl);
+
+		return (
+			<Link href={formattedURL} openInNewTab={externalUrl}>
+				{linkText}
+			</Link>
+		);
+	};
+	/**
+	TODO:
```
Comment (matthewroach) [inline blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/horizontal-links/default.jsx:54]: Is this TODO to be done with this PR? or a future task?

Comment (brentmiller1973) [reply blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/horizontal-links/default.jsx:54]: Future work....just notes for me as I move through sub tasks.  Will be removed before final PR for the entire ticket is presented

File: blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/horizontal-links/default.jsx [hunk 2]
```diff
@@ -39,6 +38,23 @@ const HorizontalLinksBar = ({ hierarchy, navBarColor, showHorizontalSeperatorDot
 	// the interior links and spans need to contrast with nav color scheme
 	const readableContrastingColor = navBarColor === "light" ? "#000" : "#fff";
 
+	const LinkBuilder = ({ item }) => {
+		const linkUrl = item.node_type === "link" ? item.url : item._id;
+		const linkText = item.node_type === "link" ? item.display_name : item.name;
+		const externalUrl = /(http(s?)):\/\//i.test(linkUrl);
```
Comment (nschubach) [inline blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/horizontal-links/default.jsx:44]: This only tests if the link has http(s) to see if it's external.  It should include a domain validation.  I think I've seen this elsewhere.  Was it copied from somewhere?

Comment (nschubach) [reply blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/horizontal-links/default.jsx:44]: Thinking this should be a utility, that's mainly my concern.

Comment (matthewroach) [reply blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/horizontal-links/default.jsx:44]: The `Link` component has this logic built into it

**Pull request merged**